en/physics/physics-raycast.md

```diff
@@ -1,92 +1,117 @@
-    const outRay = new geometry.ray();
-    geometry.ray.fromPoints(outRay, Vec3.ZERO, Vec3.UNIT_Z);
+    import { geometry, math } from "cc";
+    // Construct a ray that points to the Z-axis from the origin
+    const outRay = new geometry.Ray();
+    geometry.Ray.fromPoints(outRay, math.Vec3.ZERO, math.Vec3.UNIT_Z);
     ```
 
-3. Use the camera to construct a ray emitted from the origin of the camera to a point on the screen (or the near plane of the camera):
+3. Use the camera to construct a ray emitted from the camera origin to a point on the screen.
 
     ```ts
-    import { geometry, Camera } from 'cc';
-    const { ray } = geometry;
-    // It is assumed here that there is already a reference to cameraCom
-    const cameraCom: Camera;
-    const cameraCom: Camera;
-    // Get a ray emitted by the screen coordinates (0, 0)
-    const outRay = new ray();
-    cameraCom.screenPointToRay(0, 0, outRay);
+    import { geometry, Camera } from "cc";
+    // Take the example of a script mounted under Camera
+    const camera = this.node.getComponent(Camera);
+    // Get a ray emitted from the screen at (0, 0)
+    const outRay = new geometry;
+    camera?.screenPointToRay(0, 0, outRay);
     ```
 
-    > **Notes**:
-    > 1. You need to get a reference to a camera component or camera instance.
-    > 2. The order of the interface parameters exposed by both the camera component and the camera instance is not the same.
+    > **Note**.
+    > 1. First you need to get a reference to a camera component or camera instance.
+    > 2. The order of the interface parameters exposed by the camera component and the camera instance are different.
 
 ## Interface Introduction
 
-__Cocos Creator__ provides a set of ray detection functions. However, it should be noted that the detected object is a physics collider, and the corresponding collider component on the inspector panel, such as `BoxCollider`.
+Cocos Creator provides a set of raycast detection functions based on the physics engine.
 
-Currently, the interface is provided by PhysicsSystem, which has the following two categories:
+The interface is currently provided by [**PhysicsSystem**](__APIDOC__/en/#/docs/3.4/en/physics/classes/physicssystem.html) and has the following two classes.
 
-- `raycastAll`: Detect all colliders and return a Boolean value to indicate whether the detection was successful.
-- `raycastClosest`: Detect all colliders and return Boolean value as well.
+- `raycast` : Detects all colliders and records all detected results, obtained via `PhysicsSystem.instance.raycastResults`. The interface returns a boolean value indicating whether the detection was successful or not.
+- `raycastClosest`: detects all colliders and records the result of the detection with the shortest distance to the ray, obtained via `PhysicsSystem.instance.raycastClosestResult`. Also returns a boolean value indicating whether the detection was successful or not.
```

Comment:
"indicating whether the detection was successful or not." this is wrong.  It returns true if the ray intersects with any collider.